fix(core): ensure all initializer functions run in an injection context

crisbeto opened the pull request:
Ensures that all of the functions intended to be run in initializers are in an injection context. This is a stop-gap until we have a compiler diagnostic for it.

crisbeto commented on packages/core/test/acceptance/authoring/signal_queries_spec.ts:
```diff
@@ -31,10 +31,25 @@ describe('queries as signals', () => {
-      // non-required query results are undefined before we run creation mode on the view queries
-      const appCmpt = new AppComponent();
-      expect(appCmpt.divEl()).toBeUndefined();
+    it('should return undefined if optional query is read in the constructor', () => {
```
I had to rework these tests, because they were instantiating components manually.

devversion commented on packages/core/test/authoring/input_signal_spec.ts:
```diff
@@ -37,11 +37,17 @@ describe('input signal', () => {
+    @Component({template: ''})
+    class TestCmp {
+      input = input<number>(0);
+    }
```
`TestBed.runInInjectionContext(() => input(0))` also works and is shorter.

crisbeto replied:
Good point, I'll change the input tests but will keep the query tests as they are since they're kinda tied to a view.

devversion reviewed (approved):
LGTM. Only would optionally suggest `TestBed.runInInjectionContext` for shorter tests

crisbeto committed:
fix(core): ensure all initializer functions run in an injection context

Ensures that all of the functions intended to be run in initializers are in an injection context. This is a stop-gap until we have a compiler diagnostic for it.

crisbeto commented:
Blocked on reworking internal targets that are using the functions incorrectly.

crisbeto commented:
[Passing TGP](http://test/OCL:613904081:BASE:614998322:1710244905090:e1cf2d49), excluding some flakes and after I land a separate [cleanup CL](http://cl/615006007).

atscott commented:
This PR was merged into the repository by commit 018f8266b3ceece8144aeebe31454cd75f046381.
